examples/ffa_simulation.rs

```diff
@@ -0,0 +1,208 @@
+use hecs::*;
+use rand::{thread_rng, Rng};
+use std::io;
+
+/*
+ Simple simulation
+ Spawn multiple entities. They have health, damage, position.
+ On every tick they: 
+	1. move a little
+ 	2. Find closest entity to themself.
+ 	3. Fire at it.
+ 	4. Get damaged by other entities firing at them.
+ 	5. If health <= 0, the entities die.
+ Use print to describe state after every tick
+*/
+
+#[derive(Debug)]
+struct Position {
+	x: i32,
+	y: i32,
+}
+
+#[derive(Debug)]
+struct Health {
+	hp: i32
+}
+
+#[derive(Debug)]
+struct Speed {
+	speed: i32
+}
+
+#[derive(Debug)]
+struct Damage {
+	dmg: i32
+}
+
+#[derive(Debug)]
+struct KillCount {
+	count: i32
+}
+
+
+fn manhattan_dist(x0: i32, x1: i32, y0: i32, y1: i32) -> i32{
+	let dx = (x0 - x1).max(x1 - x0);
+	let dy = (y0 - y1).max(y1 - y0);
+	return dx + dy
+
+}
+
+fn spawn_enitites(world: &mut World, n: usize){
+
+	let mut rng = thread_rng();
+
+	for _ in 0..n {
+		let pos = Position{
+			x: rng.gen_range(-10, 10),
+			y: rng.gen_range(-10, 10),
+		};
+		let s =  Speed{speed: rng.gen_range(1, 5)};
+		let hp = Health{hp: rng.gen_range(30, 50)};
+		let dmg = Damage{dmg: rng.gen_range(1, 10)};
+		let kc = KillCount{count: 0};
+
+		world.spawn((pos, s, hp, dmg, kc));
+	}
+}
+
+fn batch_spawn_entities(world: &mut World, n: usize){
+	let mut rng = thread_rng();
+
+	let to_spawn = (0..n).map(|_|{
+
+		let pos = Position{
+			x: rng.gen_range(-10, 10),
+			y: rng.gen_range(-10, 10),
+		};
+		let s =  Speed{speed: rng.gen_range(1, 5)};
+		let hp = Health{hp: rng.gen_range(30, 50)};
+		let dmg = Damage{dmg: rng.gen_range(1, 10)};
+		let kc = KillCount{count: 0};
+
+		(pos, s, hp, dmg, kc)
+	});
+
+	world.spawn_batch(to_spawn);
+}
+
+fn move_system(world: &mut World){
```

Comment:
Maybe name this `integrate_motion` or something? Having the `_system` suffix on only one of the "systems" is a bit odd.